lib/unique_for_owner_validator.rb

```diff
@@ -13,8 +13,8 @@ def validate(record)
     end
   end
 
-  def unique_fields
-    %i(display_name)
+  def unique_fields_for(record)
```

Comment:
Did you consider putting this on the record class, instead of switching based on the type of record here? ie:

```ruby
class User
  def self.unique_fields; %i(login); end
end

record.class.unique_fields
```